spyder/api/widgets/menus.py

```diff
@@ -487,7 +488,14 @@ def showEvent(self, event):
 
         self.move(self.pos().x() + delta_x, self.pos().y())
 
-        super().showEvent(event)
+    def showEvent(self, event):
+        """Call adjustments when the menu is going to be shown."""
+        # To prevent race conditions which could cause partially showing a menu
+        # as in spyder-ide/spyder#22266 a timer is used to queue the move
+        # related events after the menu is shown. For more info you can check:
```

Comment:
```suggestion
        # To prevent race conditions which can cause partially showing a menu
        # (as in spyder-ide/spyder#22266), we use a timer to queue the move
        # related events before and after the menu is shown.
        # For more info you can check:
```